src/sdam/state/mod.rs

```diff
@@ -304,19 +379,42 @@ impl TopologyState {
     pub(crate) fn update(
         &mut self,
         server: ServerDescription,
-        options: ClientOptions,
-    ) -> Result<()> {
+        options: &ClientOptions,
+    ) -> Result<Option<TopologyDescriptionDiff>> {
+        let old_description = self.description.clone();
         self.description.update(server)?;
-        let addresses: HashSet<_> = self.description.server_addresses().cloned().collect();
 
-        for address in addresses.iter() {
+        let hosts: HashSet<_> = self.description.server_addresses().cloned().collect();
+        self.sync_hosts(&hosts, options);
+
+        Ok(old_description.diff(&self.description))
+    }
+
+    /// Start/stop monitoring tasks and create/destroy connection pools based on the new and
+    /// removed servers in the topology description.
+    ///
+    /// This must **ONLY** be called on a copy of a TopologyState, not one that is stored in a
+    /// client. The `topology` parameter should contain a reference to the Topology that
```

Comment:
I think this line about the `topology` parameter is out of date now and can be removed. ditto for `update`.

Reply:
Done